pkg/devspace/dependency/resolver.go

```diff
@@ -31,6 +31,7 @@ type ResolverInterface interface {
 // Resolver implements the resolver interface
 type resolver struct {
 	DependencyGraph *graph.Graph
```

Comment:
I think you may be able to use the DependencyGraph to avoid re-downloading / re-pulling dependencies.